pkg/bff/api/v1/client_test.go

```diff
@@ -172,36 +170,143 @@ func TestRegister(t *testing.T) {
-		TRISAEndpoint:    "trisa.example.com:443",
-		CommonName:       "trisa.example.com",
-		Website:          "https://example.com",
-		BusinessCategory: "PRIVATE_ORGANIZATION",
-		VASPCategories:   []string{"ATM", "Other"},
-		EstablishedOn:    "2019-01-14",
-	}
-
-	out, err := client.Register(context.TODO(), req)
+	out, err := client.VerifyContact(context.TODO(), &api.VerifyContactParams{Directory: "trisatest.net", ID: "foo", Token: "bar"})
 	require.NoError(t, err)
-	require.Equal(t, fixture.Id, out.Id)
-	require.Equal(t, fixture.RegisteredDirectory, out.RegisteredDirectory)
-	require.Equal(t, fixture.CommonName, out.CommonName)
 	require.Equal(t, fixture.Status, out.Status)
 	require.Equal(t, fixture.Message, out.Message)
-	require.Equal(t, fixture.PKCS12Password, out.PKCS12Password)
 }
 
-func TestVerifyContact(t *testing.T) {
-	fixture := &api.VerifyContactReply{
-		Status:  "PENDING_REVIEW",
-		Message: "thank you for verifying your email",
-	}
+func TestLogin(t *testing.T) {
+	// Test login with credentials and a TLS connection sets csrf protection
+	var err error
+	token := api.Token("eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiIxMjM0NTY3ODkwIiwibmFtZSI6IkpvaG4gRG9lIiwiaWF0IjoxNTE2MjM5MDIyfQ.SflKxwRJSMeKKF2QT4fwpMeJf36POk6yJV_adQssw5c")
+
+	// Create a Test TLS Server
+	ts := httptest.NewTLSServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if r.Method != http.MethodPost {
+			w.WriteHeader(http.StatusMethodNotAllowed)
+			return
+		}
+
+		if r.URL.Path != "/v1/users/login" {
+			w.WriteHeader(http.StatusNotFound)
+			return
+		}
+
+		if r.Header.Get("Authorization") != "Bearer "+string(token) {
+			w.WriteHeader(http.StatusUnauthorized)
+			return
+		}
+
+		// Set double cookies
+		cookie := http.Cookie{
+			Name:     "csrf_token",
+			Value:    "thisisanexamplecookietoken",
+			MaxAge:   600,
+			Secure:   true,
+			HttpOnly: false,
+			Path:     "/",
+		}
+		http.SetCookie(w, &cookie)
+
+		cookie.Name = "csrf_reference_token"
+		cookie.HttpOnly = true
+		http.SetCookie(w, &cookie)
+
+		w.WriteHeader(http.StatusNoContent)
+	}))
+	defer ts.Close()
+
+	// Fetch the https client and add a cookie jar
+	client := ts.Client()
+	client.Jar, err = cookiejar.New(nil)
+	require.NoError(t, err, "couldn't add a cookie jar to the https client")
+
+	// Create the BFF api client
+	bff, err := api.New(ts.URL, api.WithClient(client), api.WithCredentials(token))
+	require.NoError(t, err, "couldn't create BFF client with https and credentials")
+
+	// Execute the Login request
+	err = bff.Login(context.TODO())
+	require.NoError(t, err, "could not login using the bff client")
+
+	// Check to ensure double cookies are set. This doesn't test our code, but ensures
+	// that tests that depend on double cookies will work in the future.
+	u, err := url.Parse(ts.URL)
+	require.NoError(t, err, "could not parse test server url")
+	require.Len(t, client.Jar.Cookies(u), 2, "expected two cookies set in the cookie jar")
```

Comment:
@pdeziel verified that `Login` works to set double cookies as long as we use `httptest.NewTLSServer` … the question is how to do this in the bff tests …